src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +137,34 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+
+    /**Gets the Server Information
+     *
+     * Based on request and response time.
+     *
+     * @return null
+     */
+  public UptimeServerInfo getInfo(){
+    try(final Connection connection = source.connect()){
+        Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+        if(Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE){
+            final Time startTime = Time.SERIALIZER.read(connection.in());
+            return new UptimeServerInfo(startTime, version);
```

Comment:
I would add a comment here as to why this is creating a new UpTimeServerInfo with both a startTime and a version. I am not sure why this is necessary as you have a separate VersionServerInfo class.